File: MB-Secretaria/providers/local/adapter.ts
import { buildProviderResponse } from "../provider-response";
import type { ProviderRequest, ProviderResponse } from "../types";
import { LOCAL_CAPABILITIES } from "./capabilities";
import { localHealthDisabled } from "./health";
import type { LocalProviderAdapter } from "./contract";

export class DisabledLocalAdapter implements LocalProviderAdapter {
  readonly provider_name = "local" as const;
  readonly model_name = "local-disabled";
  readonly capabilities = LOCAL_CAPABILITIES;

  async complete(input: ProviderRequest): Promise<ProviderResponse> {
    return buildProviderResponse({
      request: input,
      provider_name: "local",
      model_name: this.model_name,
      status: "disabled",
      safety_flags: ["PROVIDER_DISABLED"],
    });
  }

  async healthcheck() {
    return localHealthDisabled();
  }
}

File: MB-Whatsapp/providers_py/local/adapter.py
from __future__ import annotations

from ..provider_health import disabled_provider_health
from ..provider_response import build_provider_response
from ..types import ProviderAdapter, ProviderCapabilities, ProviderRequest, ProviderResponse


class DisabledLocalAdapter(ProviderAdapter):
    def __init__(self) -> None:
        super().__init__(
            provider_name="local",
            model_name="local-disabled",
            capabilities=ProviderCapabilities(True, False, False, False, False, False, 2048, True),
        )

    def complete(self, request: ProviderRequest) -> ProviderResponse:
        return build_provider_response(request, provider_name="local", model_name=self.model_name, status="disabled", safety_flags=["LOCAL_DISABLED"])

    def healthcheck(self):
        return disabled_provider_health("local")
